src/Persistence/Sql.php

```diff
@@ -971,6 +971,19 @@ private function getIdSequenceName(Model $model): ?string
 
     public function lastInsertId(Model $model): string
     {
+        // TODO: Oracle does not support lastInsertId(), only for testing
+        // as this does not support concurrent inserts
+        if ($this->connection instanceof \atk4\dsql\Oracle\Connection) {
```

Comment:
Same here - we should move this to sql Connection/Expression classes

Reply:
this can be moved easier, but we will have to still provide a lot of data about the Model (but not Model, as we are in dsql there)